packages/edit-navigation/src/store/resolvers.js

```diff
@@ -115,7 +112,19 @@ function createNavigationBlock( menuItems ) {
 	return [ navigationBlock, menuItemIdToClientId ];
 }
 
-function convertMenuItemToLinkBlock( menuItem, innerBlocks = [] ) {
+function convertMenuItemToBlock( menuItem, innerBlocks = [] ) {
+	if ( menuItem.type === 'html' ) {
+		const parsedBlocks = parse( menuItem.content.raw );
+
+		if ( parsedBlocks.length !== 1 ) {
```

Comment:
This will kick in once a block plugin is disabled. I don't think there's a good way of addressing it, but just wanted to double check - is this behavior consistent with the post editor?

Reply:
If a block plugin is disabled then `parse()` will return an array containing one `core/missing` block. The body of this `if()` won't be executed and so a `core/missing` is what will be used for the menu item. This matches the behaviour of the post editor.

The `if()` here is to guard against the case where, for whatever reason, there is 0 or >1 blocks in the menu item's HTML. I'm not really sure what to do in this case—it's undefined behaviour and probably indicates a bug somewhere. Perhaps failing loudly would be better?

Reply:
I put it on my todo list to come back to this as will need to think on it further.